cmd/operator-opamp-bridge/config/config.go

```diff
@@ -16,43 +16,99 @@ package config
 
 import (
 	"crypto/rand"
+	"errors"
 	"fmt"
+	"io/fs"
+	"net/url"
 	"os"
 	"runtime"
 	"time"
 
+	"github.com/go-logr/logr"
 	"github.com/oklog/ulid/v2"
-	"github.com/open-telemetry/opamp-go/client"
+	opampclient "github.com/open-telemetry/opamp-go/client"
 	"github.com/open-telemetry/opamp-go/protobufs"
+	"github.com/spf13/pflag"
 	"gopkg.in/yaml.v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
 
+	"github.com/open-telemetry/opentelemetry-operator/apis/v1alpha1"
 	"github.com/open-telemetry/opentelemetry-operator/cmd/operator-opamp-bridge/logger"
 )
 
 const (
-	agentType             = "io.opentelemetry.operator-opamp-bridge"
-	defaultConfigFilePath = "/conf/remoteconfiguration.yaml"
+	agentType = "io.opentelemetry.operator-opamp-bridge"
 )
 
 var (
-	agentVersion = os.Getenv("OPAMP_VERSION")
-	hostname, _  = os.Hostname()
+	agentVersion  = os.Getenv("OPAMP_VERSION")
+	hostname, _   = os.Hostname()
+	schemeBuilder = k8sruntime.NewSchemeBuilder(registerKnownTypes)
+)
+
+func registerKnownTypes(s *k8sruntime.Scheme) error {
+	s.AddKnownTypes(v1alpha1.GroupVersion, &v1alpha1.OpenTelemetryCollector{}, &v1alpha1.OpenTelemetryCollectorList{})
+	metav1.AddToGroupVersion(s, v1alpha1.GroupVersion)
+	return nil
+}
+
+func GetLogger() logr.Logger {
+	return zap.New(zap.UseFlagOptions(&zapCmdLineOpts))
+}
+
+type Capability string
+
+const (
```

Comment:
nit: is there an approach to the order of these consts? I think alpha-ordering might be easier to look at and review

Reply:
this came from the upstream opamp-go ordering, but agreed... it's a bit odd.